Maps/ArchipelagoCampaign/WoL/ap_a_sinister_turn.SC2Map/MapScript.galaxy

```diff
@@ -4651,6 +4659,14 @@ bool gt_DialogueRoboticsFacilityQ_Func (bool testConds, bool runActions) {
     MinimapPing(PlayerGroupAll(), PointFromId(190), 6.0, Color(0.00, 100.00, 0.00));
     UIAlertPoint("Trigger", gv_p1_USER, StringExternal("Param/Value/9860ACE5"), null, PointFromId(190));
     Wait(1.0, c_timeReal);
+    auto5F805BF8_g = UnitGroup("AP_ImmortalAiur", gv_p1_USER, null, UnitFilter(0, 0, (1 << c_targetFilterMissile), (1 << (c_targetFilterDead - 32)) | (1 << (c_targetFilterHidden - 32))), 0);
+    auto5F805BF8_u = UnitGroupCount(auto5F805BF8_g, c_unitCountAll);
+    for (;; auto5F805BF8_u -= 1) {
+        auto5F805BF8_var = UnitGroupUnitFromEnd(auto5F805BF8_g, auto5F805BF8_u);
+        if (auto5F805BF8_var == null) { break; }
+        libNtve_gf_SendActorMessageToUnit(auto5F805BF8_var, "Destroy Immediate");
+        ActorCreate(ActorScopeFromUnit(auto5F805BF8_var), "AP_ImmortalAiur", null, null, null);

```

Comment:
Refreshing the Actor won't help, you need to replace the unit with same type after rescue

Reply:
I _thought_ it was a little weird, yeah... all I saw when I looked at the Harvest of Screams rescue behavior was the refresh actor trigger though, so I just shrugged and went along with it. Will try the full-replace trigger instead later today.